src/github/api/mod.rs

```diff
@@ -0,0 +1,367 @@
+mod read;
+mod write;
+
+use crate::utils::ResponseExt;
+use anyhow::{bail, Context};
+use hyper_old_types::header::{Link, RelationType};
+use log::{debug, trace};
+use reqwest::header::HeaderMap;
+use reqwest::{
+    blocking::{Client, RequestBuilder, Response},
+    header::{self, HeaderValue},
+    Method, StatusCode,
+};
+use serde::{de::DeserializeOwned, Deserialize};
+use std::borrow::Cow;
+use std::fmt;
+
+pub(crate) use read::GitHubRead;
+pub(crate) use write::GitHubWrite;
+
+struct HttpClient {
+    client: Client,
```

Comment:
Would it be feasible to have a read/write flag on `HttpClient` that causes requests that look like writes (e.g., DELETE, POST, etc. requests) to fail unless the write flag is set? This could one extra layer protecting against accidentally writing when we don't mean to. I think the only challenge with this is that we don't have good insight into whether a graphql query is a write.

Reply:
Good idea! I added a new commit with this.